EKF/ekf_helper.cpp

```diff
@@ -271,6 +271,9 @@ void Ekf::resetHeight()
 			_state.pos(2) = _hgt_sensor_offset - gps_newest.hgt + _gps_alt_ref;
 
 			// the state variance is the same as the observation
+			// TODO: the vacc is not constrained using the parameters
+			// do we really want to use the latest and not delayed measurement?
```

Comment:
@priseborough Since the main EKF runs on delayed-time, why are we using the newest measurement here?

Reply:
If the reset to a sensor is guaranteed to be actioned on the same frame the sensor data is pulled from the buffer, the using the delayed data makes more sense. The exception to this is when the reset can be triggered at any time, in which case the data from the buffer can be old by up to the measurement interval, in which case using the latest data from the buffer input can result in smaller worst case timing errors. For example with GPS at 5Hz (200msec) and a buffer 100msec long, taking data from the input to the buffer results in a timing error between -100 and +100 msec, whereas if the output data was used, the timing error would be between 0 and 200msec.

Reply:
Ok, thanks for the explanation, that makes sense now